kernel/cpu/src/aarch64.rs

```diff
@@ -52,15 +52,16 @@ impl MpidrValue {
     /// Reads an affinity `level` from this `MpidrValue`.
     ///
     /// Panics if the given affinity level is not 0, 1, 2, or 3.
-    pub fn affinity(self, level: u8) -> u8 {
+    pub fn affinity(self, level: u8) -> u64 {
```

Comment:
you can avoid all of these panics if you create a new enum and use that as the argument instead of a raw `u8`.

ex:
```rust
pub enum MpidrAffinity {
    Zero = 0,
    One  = 8,
    Two  = 16,
    Three = 32,
}
```
note that if you need the values of each variant to be the integers 0, 1, 2, 3, then you can do that too.